### PR TITLE
Bump rxjava to version 3

### PR DESCRIPTION
**Pull request by luizgrp**
https://github.com/vanniktech/RxRiddles/pull/27 can be discarded if this gets merged first

### CODE REVIEW

**Comment by vanniktech** [comment]
Can you resolve conflicts?

**Comment by luizgrp** [comment]
done 👍

File: src/test/kotlin/com/vanniktech/rxriddles/Riddle102Test.kt [hunk 1]
```diff
@@ -43,4 +44,14 @@ class Riddle102Test {
 
     o.assertResult(10 to 5)
   }
+
+  private fun <T> TestObserver<T>.awaitTerminalEvent(): Boolean {
```
Comment (vanniktech) [inline src/test/kotlin/com/vanniktech/rxriddles/Riddle102Test.kt:48]: This got removed in rx3? If so, can you pull it into it's own file? (If there's another extension file in the test directory you can place it there as well)

Comment (luizgrp) [reply src/test/kotlin/com/vanniktech/rxriddles/Riddle102Test.kt]: yup, see [notes](https://github.com/ReactiveX/RxJava/wiki/What's-different-in-3.0#test-support-methods)

Comment (luizgrp) [reply src/test/kotlin/com/vanniktech/rxriddles/Riddle102Test.kt]: let me know if you are happy with the file name

**Pull request merged**